src/Event/BaseEvent.php

```diff
@@ -77,9 +77,10 @@ abstract public function getEventType();
     /**
      * Returns the array to be used for encoding this object as json.
      *
-     * @return array
+     * @return mixed
      */
-    public function jsonSerialize(): mixed
+    #[\ReturnTypeWillChange]
+    public function jsonSerialize()
```

Comment:
I think this is proper way to deal with deprecation notices in php 8 while maintaining BC.